### ISSUE TITLE
Whitelist problem

### ISSUE THREAD

**Issue by catcun** [mailcow/mailcow-dockerized#651]
Hello,
I have problems with antispam. I receive lots of notifications from a bank but as most of the time they are rejected. So I added the sender email address to the domain’s whitelist and it works. But after some time it stops working and I need to restart mailcow’s services. I don’t remember which services exactly as I am restarting most but I believe that redis-mailcow rspamd-mailcow are needed to be restarted.  
The issue is weird because sometimes I restart the whole server to apply kernel updates and the email is rejected; but after restarting mailcow services the email is whitelisted.

To be clear: A whitelisted email is being rejected by the antispam engine and that’s the main issue here.
[postfix_logs.txt](https://github.com/mailcow/mailcow-dockerized/files/1356737/postfix_logs.txt)
[rspamd_logs.txt](https://github.com/mailcow/mailcow-dockerized/files/1356736/rspamd_logs.txt)

**Comment by catcun** [comment]
I still have no clue on why is the whitelist being ignored. But I see that FUZZY_DENIED is giving a high score to the email messages from the bank:
FUZZY_DENIED(10.514324)[1:dbf46bbce3:1.00:bin,1:7773a9ecd0:1.00:txt]
After investigation, I see that mailcow is using rspamd.com fuzzy storage so there is no way for me to delete the hash using fuzzy_del. If I am right I have 2 options:

- Setup my own storage. At least for now this is way over my head.

- Somehow whitelist the messages/sender/hash

I tried the whitelist option by editing data/conf/rspamd/local.d/rspamd.conf.local like this:
```
# rspamd.conf.local
history_redis {}
worker "log_helper" {
  count = 1;
}
settings {
	notificaciones_xyz {
		priority = high;
		from = " xyz";
		apply "default" {
			FUZZY_DENIED = 0.0;
		}
	}
}
```
And restarted rspamd. With no luck. Does anyone here know how I could whitelist the messages with fuzzy_check?
Thank you

**Comment by andryyy** [comment]
Can you forward me a mail from xyz as attachment? As long as there is no personal information in it. 👍  info@servercow.de

**Comment by mkuron** [comment]
There‘s something wrong with force_action on the development version of rspamd, see https://github.com/vstakhov/rspamd/pull/1846 . It’s possible that this is causing your whitelisting issues.

Half of the bug is fixed (though Mailcow‘s docker container hasn‘t been rebuilt yet), while the other half is still broken.

**Comment by catcun** [comment]
@andryyy Sent you an example
@mkuron I adjusted my score to 0.1; I will see if that helps.
Thank you both; I will keep investigating and report back.

**Comment by mkuron** [comment]
I've confirmed that this is fixed with #666. So the next time @andryyy merges dev to master and you run the update, the problem will be fixed for you. In the meantime, you can also manually apply this fix by updating rspamd:

```bash
docker-compose build rspamd-mailcow
docker-compose up -d
```

**Comment by catcun** [comment]
@mkuron Thank you. I updated rspamd image but fuzzy_check is still adding a high score to the messages. I guess I will need to disable fuzzy_check

**Comment by mkuron** [comment]
@catcun, whitelist has a score of -999, which should be enough to compensate for anything te fuzzy check adds. Do you see MAILCOW_WHITE being applied to the message?

**Comment by catcun** [comment]
@mkuron No MAILCOW_WHITE at all. And I have the email address xyz in the whitelist for the domain: 

FUZZY_DENIED(10.514324)[1:dbf46bbce3:1.00:bin,1:7773a9ecd0:1.00:txt]
HFILTER_HELO_IP_A(1)[ironport2.com]
HFILTER_HELO_NORES_A_OR_MX(0.3)[ironport2.com]
DMARC_POLICY_ALLOW(-0.25)[xyz,reject]
R_SPF_ALLOW(-0.2)[+mx]
R_DKIM_ALLOW(-0.2)[xyz]
MIME_GOOD(-0.1)[multipart/mixed,multipart/related,text/plain]
BAYES_SPAM(0.009289)[43.31%]
DYN_RL_CHECK(0)
DKIM_TRACE(0)[xyz:+]
RCPT_MAILCOW_DOMAIN(0)[xxxxx.com]
RCVD_TLS_ALL(0)
GREYLIST(0)[pass,meta]
IP_SCORE(0)[ip: (7.17), ipnet: 170./24(3.54), asn: 2134(2.48), country: ES(-0.02)]
ASN(0)[asn:2134, ipnet:170./24, country:ES]
FROM_EQ_ENVFROM(0)
TO_MATCH_ENVRCPT_ALL(0)
ARC_NA(0)
RCVD_IN_DNSWL_MED(0)[.170.list.dnswl.org : 127.0.2.2]
FROM_HAS_DN(0)
RCPT_COUNT_ONE(0)[1]
RCVD_COUNT_ZERO(0)[0]
TO_DN_EQ_ADDR_ALL(0)

Before updating to master I was able to restart the container and MAILCOW_WHITE was applied temporarily until I had to restart the container again.
I just tried to disable fuzzy_check  in local.d/options.inc to see if that allows the messages

**Comment by andryyy** [comment]
Are you sure you added `xyz`? You sent me two msgs where one was whitelisted (xyz) while the other had no such symbol (xyz).

**Comment by catcun** [comment]
@andryyy Exactly, in one message the whitelist worked in the other it didn't. I have xyz and xyz in the whitelist for the domain.
xyz is there because of #650

**Comment by catcun** [comment]
Follow up: I disabled fuzzy_check in local.d/options.inc and messages are now getting in

**Comment by catcun** [comment]
@andryyy Thank you for all of your support and wonderful mailcow!.
@mkuron It seems like your work with rspamd has paid off for me. Thanks!

Everything is working now

I promised a follow up but waited until I was able to fully confirm that everything is working.
After the last update I re-enabled fuzzy_check with a very low FUZZY_DENIED score for the problematic email address. I can confirm that fuzzy_check is working and the previously problematic emails are not being blocked by it.

Thanks!
